client/marketing/components/knowledge-base/ReadBlogMessage.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import { _x } from '@wordpress/i18n';
+
+const ReadBlogMessage = () => {
+	const message1 = _x(
+		'Read ',
+		'Read <a href="https://woocommerce.com/blog/marketing/coupons/">the WooCommerce blog</a> for more tips on marketing your store',
+		'woocommerce-admin'
+	);
+
+	const message2 = _x(
+		'the WooCommerce blog',
+		'Read <a href="https://woocommerce.com/blog/marketing/coupons/">the WooCommerce blog</a> for more tips on marketing your store',
+		'woocommerce-admin'
+	);
+
+	const message3 = _x(
+		' for more tips on marketing your store',
+		'Read <a href="https://woocommerce.com/blog/marketing/coupons/">the WooCommerce blog</a> for more tips on marketing your store',
+		'woocommerce-admin'
+	);
+
```

Comment:
There's a lot of repetition going on here, you could probably do something like:

```js
return interpolateComponents(
    mixedString: __( 'Read {{link}}the WooCommerce blog{{/link}} for more tips on marketing your store', 'woocommerce-admin' ),
    components: {
        link: <Link href="https://woocommerce.com/blog/marketing/coupons/" />
    }
);
```

See https://github.com/woocommerce/woocommerce-admin/blob/main/client/profile-wizard/steps/benefits/index.js#L259 for more of an example.

Reply:
If you can get the size of this function down you could probably inline it as well to avoid having an extra file.

Reply:
@becdetat , thanks for the heads up! I'll read more about the `interpolateComponents`.

> you could probably inline it as well to avoid having an extra file.

My personal preference is to favor many small files over few large files. Is that the general rule of thumb / preference for the project here? If that's the case, I'll work with that.

Reply:
My opinion is that you can go too far, if something can be written succinctly inline then it is more valuable being co-located where it is used.

If this was something that was going to be reused in multiple places then definitely go with a file per "thing" but in this case it should be inline.